src/main/java/com/questhelper/helpers/mischelpers/farmruns/PaymentTracker.java

```diff
@@ -0,0 +1,196 @@
+	private final FarmingWorld farmingWorld;
+
+	private int lastSelectedOption;
+
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		Widget text = client.getWidget(ComponentID.DIALOG_NPC_TEXT);
+		if (text == null || !PAYMENT_TEXT.contains(text.getText()))
+		{
+			return;
+		}
+
+		Widget name = client.getWidget(ComponentID.DIALOG_NPC_NAME);
+		Widget head = client.getWidget(ComponentID.DIALOG_NPC_HEAD_MODEL);
+		if (name == null || head == null || head.getModelType() != WidgetModelType.NPC_CHATHEAD)
+		{
+			return;
+		}
+
+		final int npcId = head.getModelId();
+		final FarmingPatch patch = findPatchForNpc(npcId);
+		if (patch == null)
+		{
+			return;
+		}
+
+		if (getProtectedState(patch))
+		{
+			return;
+		}
+
+		log.debug("Detected patch payment for {} ({}) patch {}", name.getText(), npcId, patch);
+		setProtectedState(patch, true);
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked opt)
+	{
+		var action = opt.getMenuAction();
+		// look for resume_pausebutton from click
+		if (action == MenuAction.WIDGET_CONTINUE)
+		{
+			var w = opt.getWidget();
+			if (w != null && w.getId() == ComponentID.DIALOG_OPTION_OPTIONS && w.getIndex() > -1 && isPatchOption(w.getText()))
+			{
+				lastSelectedOption = w.getIndex() - 1; // subid 0 is "Select an Option"
+				log.debug("Selected option via click: {}", lastSelectedOption);
+			}
+		}
+		// look for selecting the Pay NPC option
+		else if ((action == MenuAction.NPC_THIRD_OPTION || action == MenuAction.NPC_FOURTH_OPTION) && opt.getMenuOption().startsWith("Pay"))
+		{
+			lastSelectedOption = action == MenuAction.NPC_THIRD_OPTION ? 0 : 1;
+			log.debug("Selected option via npc op: {}", lastSelectedOption);
+		}
+	}
+
+	@Subscribe
+	public void onScriptPreFired(ScriptPreFired scriptPreFired)
+	{
+		// look for resume_pausebutton from keypress
+		if (scriptPreFired.getScriptId() == ScriptID.CHATBOX_KEYINPUT_MATCHED)
+		{
+			int[] intStack = client.getIntStack();
+
+			int componentId = intStack[0];
+			int subId = intStack[1];
+
+			var w = client.getWidget(componentId).getChild(subId);
+			if (componentId == ComponentID.DIALOG_OPTION_OPTIONS && subId > -1 && isPatchOption(w.getText()))
+			{
+				lastSelectedOption = subId - 1; // subid 0 is "Select an Option"
+				log.debug("Selected option via keypress: {}", lastSelectedOption);
+			}
+		}
+	}
+
+	private static boolean isPatchOption(String name)
+	{
+		if (name == null)
+		{
+			return false;
+		}
+
+		return name.contains("Patch") || name.contains("allotment");
+	}
+
+	private static String configKey(FarmingPatch fp)
+	{
+		return fp.configKey() + "." + TimeTrackingConfig.PROTECTED;
+	}
+
+	void setProtectedState(FarmingPatch fp, boolean state)
+	{
+		if (!state)
+		{
+			configManager.unsetRSProfileConfiguration(TimeTrackingConfig.CONFIG_GROUP, configKey(fp));
+		}
+		else
+		{
+			configManager.setRSProfileConfiguration(TimeTrackingConfig.CONFIG_GROUP, configKey(fp), state);
+		}
+	}
```

Comment:
I don't think our copy of this file should be making modifications to the time tracking config.
If we want to store our own copy of it, it should be in our own config group.

If possible, it'd be easiest if we can rely on reading TimeTrackingConfig's config value

Reply:
Good spot, I've removed any setting for it now. This means this helper won't work if the time tracking plugin isn't running, but that's probably unlikely for most people? May be worth making our own handling but really I'd want to rely on the time plugin over duplicating work.